controller/appcontroller.go

```diff
@@ -490,23 +501,27 @@ func (ctrl *ApplicationController) finalizeApplicationDeletion(app *appv1.Applic
 		if !apierr.IsNotFound(err) {
 			logCtx.Errorf("Unable to get refreshed application info prior deleting resources: %v", err)
 		}
-		return nil
+		return nil, nil
 	}
 
 	objsMap, err := ctrl.stateCache.GetManagedLiveObjs(app, []*unstructured.Unstructured{})
 	if err != nil {
-		return err
+		return nil, err
 	}
 	objs := make([]*unstructured.Unstructured, 0)
 	for k := range objsMap {
-		if objsMap[k].GetDeletionTimestamp() == nil && shouldBeDeleted(app, objsMap[k]) {
+		permitted, err := ctrl.isPermittedForDelete(app, objsMap[k])
```

Comment:
do you want to move this into the block below?

Reply:
Do you mean like instead of
```go
		permitted, err := ctrl.isPermittedForDelete(app, objsMap[k])
		if err != nil {
			return nil, err
		}
		if permitted && objsMap[k].GetDeletionTimestamp() == nil && shouldBeDeleted(app, objsMap[k]) {
			objs = append(objs, objsMap[k])
		}
```
something along these lines?
```go
			permitted, err := ctrl.isPermittedForDelete(app, objsMap[k])
			if err != nil {
				return nil, err
			} else if permitted {
				objs = append(objs, objsMap[k])
			}
```